### PR TITLE
cr provider can update mg slice fields

### PR DESCRIPTION
**Pull request by fahedouch**
Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### Description of your changes
the crossplane observation mechanism must allow the addition of data in the slices of managed resources 

related comment # https://github.com/crossplane/provider-gcp/pull/400#issuecomment-955672395

I have:

- [x ] Read and followed Crossplane's [contribution process].
- [x ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

nothing to report

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/reconciler/managed/reconciler.go [hunk 1]
```diff
@@ -311,8 +311,8 @@ type ExternalObservation struct {
 	// ResourceLateInitialized should be true if the managed resource's spec was
 	// updated during its observation. A Crossplane provider may update a
 	// managed resource's spec fields after it is created or updated, as long as
-	// the updates are limited to setting previously unset fields, and adding
-	// keys to maps. Crossplane uses this information to determine whether
+	// the updates are limited to setting previously unset fields, adding
+	// keys to maps and adding data to slices. Crossplane uses this information to determine whether
```
Comment (negz) [inline pkg/reconciler/managed/reconciler.go:315]: Nit: Could you please wrap this comment at 80 cols.

Comment (fahedouch) [reply pkg/reconciler/managed/reconciler.go]: @negz done

**Comment by fahedouch** [comment]
@negz after reviewing this issue , can we have your opinion on  this ticket please :  https://github.com/crossplane/provider-gcp/pull/400 . It is related to the actual issue.

**Comment by negz** [comment]
I believe the docstring is accurate as is - we should not late init slice elements. See my comment at https://github.com/crossplane/provider-gcp/pull/400#issuecomment-983146711 for more context.

**Pull request closed without merging**